src/ModelManager.ts

```diff
@@ -141,11 +141,8 @@ export class ModelManager {
         if (this._modelClient) {
             domain = this._modelClient.apiHost != null ? this._modelClient.apiHost : "";
         }
-        // Check if async is required. This will be true only in remote rendering.
-        const asyncFlag =  domain === PathUtils.getCurrentPathname() ? false : true;
-        if (asyncFlag && Utils.isEditMode()) {
-            const clientLibUrl = Utils.generateClientLibsUrl(rootModelPath, domain);
-            Utils.appendClientLibs(clientLibUrl);
+        if (Utils.isAsync(domain) && Utils.isEditMode()) {
+            Utils.appendClientLibs(domain);
```

Comment:
Should we expose the conditions in the calling function? Or should we internalize the conditions in `Utils.appendClientLibs` and call it without condition?

Reply:
I think calling this after conditions make it more understandable(personally). But of course, we can move the condition inside.